crates/runtime/src/model/tool_use.rs

```diff
@@ -223,6 +224,20 @@ impl ToolUsingChat {
         messages.push(assistant_message);
         messages.extend(tool_messages);
 
+        if !messages.is_empty() {
+            // exclude "list_datasets" tool from counting, since it is used on every AI inference call
+            let used_tools = spiced_tools
+                .iter()
+                .any(|t| t.function.name != "list_datasets");
+
+            if used_tools {
+                let context = RequestContext::current(AsyncMarker::new().await);
+                if let Some(model_context) = context.extension::<ModelContextExtension>().await {
+                    model_context.set_tools_used(true);
+                }
```

Comment:
consider adding a panic in dev mode, to make sure we include it in all places where it is supposed to be included

Reply:
Added